src/main/java/pages/FlightSearchPage.java

```diff
@@ -0,0 +1,235 @@
+            waitForCalendarUpdate();
+        }
+
+        if (!isTargetMonthVisible(targetMonth)) {
+            throw new NoSuchElementException("Target month not visible after max retries: " + targetMonth);
+        }
+
+        WebElement correctCalendar = getCalendarForMonth(targetMonth);
+        WebElement dateCell = correctCalendar.findElement(By.xpath(String.format(CALENDAR_DATE_CELL, day)));
+        scrollIntoView(dateCell);
+        dateCell.click();
+    }
+
+    public void selectReturnDate(String day) {
+        WebElement returnInput = getReturnInput();
+        jsClick(returnInput);
+        wait.until(ExpectedConditions.visibilityOf(calendarRight));
+        WebElement dateCell = calendarRight.findElement(By.xpath(String.format(CALENDAR_DATE_CELL, day)));
+        scrollIntoView(dateCell);
+        dateCell.click();
+    }
+
+    public void selectPassengers(int adults, int children, int babies) {
+        wait.until(ExpectedConditions.elementToBeClickable(passengersButton));
+        scrollIntoView(passengersButton);
+        jsClick(passengersButton);
+        wait.until(ExpectedConditions.visibilityOf(passengerPopover));
+
+        int currentAdults = Integer.parseInt(adultInput.getAttribute("value"));
+        while (currentAdults < adults) {
+            adultPlusButton.click();
+            currentAdults++;
+        }
+
+        int currentChildren = Integer.parseInt(childInput.getAttribute("value"));
+        while (currentChildren < children) {
+            childPlusButton.click();
+            currentChildren++;
+        }
+
+        int currentBabies = Integer.parseInt(babyInput.getAttribute("value"));
+        while (currentBabies < babies) {
+            babyPlusButton.click();
+            currentBabies++;
+        }
+
+        wait.until(ExpectedConditions.elementToBeClickable(applyPassengersButton)).click();
+        System.out.printf("Passengers selected — Adults: %d, Children: %d, Babies: %d%n", adults, children, babies);
+    }
+
+    public void selectCurrency(String currencyCode) {
+        wait.until(ExpectedConditions.elementToBeClickable(currencyDropdown));
+        jsClick(currencyDropdown);
+        wait.until(ExpectedConditions.visibilityOf(dropdownMenu));
+        WebElement currencyOption = wait.until(ExpectedConditions.elementToBeClickable(
+                By.xpath(String.format(CURRENCY_OPTION, currencyCode))));
+        currencyOption.click();
+    }
+
+    private boolean isTargetMonthVisible(String targetMonth) {
+        for (WebElement header : calendarHeaders) {
+            if (header.getText().toLowerCase().contains(targetMonth.toLowerCase())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private WebElement getCalendarForMonth(String targetMonth) {
+        for (WebElement calendar : calendarContainers) {
+            WebElement header = calendar.findElement(By.cssSelector("th.month"));
```

Comment:
Don't hardcode locators inside methods. This element should be initialized using PageFactory